### PR TITLE
V1/53 54 subscribe unsubscribe frontend

### LINKED ISSUE #53: Join presentation
Create vue.js component for joining a presentation

## User Story

As a Authenticaded user
I want to have a GUI to subscribed to a presentaion
So that allow me to see content of presentation

- [ ] define what must be on the component
- [ ] define the style (follow the styleguide)
- [ ] add communication to the backend
- [ ] test the component 

**Estimation: 2h**

### LINKED ISSUE #54: Leave presentation
Create vue.js component for leaving a presentation

As a user
I want to have a GUI to unsubscribe from a presentation
So i am not participating to that presentation anymore

**Estimation: 1h**

### PR DESCRIPTION
**Pull request by fabio10000**
# How to test
## Route
in the router.js make sure to have the route `/presentation/:idPresentation`

## Login
make sure to have the access token saved on your storage with a key named Authorization-token

## Presentation User relations
Don't hesitate to change values on the db to check if the correct button is shown.

## Cannot be tested
On this issue the real backend for subscription and unsubscription is not working yet so we cannot test but the real API route is used (cf: API reference doc) so it should work

### CODE REVIEW

**Review by SundayBraid — COMMENTED** [review]
Test report
- [x] user unsubscribed can successfully sub to a presentation
- [x] user subscribed an unsubscribed can successfully unsub
- [x] HTTP request are succefully sent when hitting button
- [x] feedback message are present
- [x] button is updating when hitting on it (sub<->unsub)

## not tested
Interaction with backend as it is explain it cannot be tested in the PR comment

File: src/resources/js/components/Presentation.vue [hunk 1]
```diff
@@ -64,15 +63,8 @@ export default {
...
   },
   beforeMount() {
-    // set the bearer token
-    window.axios.defaults.headers.common = { Authorization: `Bearer ${localStorage.getItem('Authorization-token')}` };
-
-    /** *************************************************************************
-     * there is an issue with the route here, it appears that route
-     * with the following pattern "aaaa/bbbb" lead to a 404 error
-     */
+
```
Comment (SundayBraid) [inline src/resources/js/components/Presentation.vue]: there is a linting issue after the merge, there is an extra line space here (line 66)

**Review by SundayBraid — CHANGES_REQUESTED** [review]

File: src/resources/js/components/Presentation.vue (hunk 1 above)
Comment (fabio10000) [reply src/resources/js/components/Presentation.vue]: fixed

**Review by SundayBraid — APPROVED** [review]

File: src/resources/js/components/Presentation.vue [hunk 2]
```diff
@@ -47,20 +63,32 @@ export default {
...
+      });
+
+    // get current user
+    window.axios
+      .get('/me')
+      .then((userResponse) => {
+        this.user = userResponse.data.data;
+        // check if the user is subscribed
+        window.axios
+          .get(`/users/${this.user.id}/presentations`)
```
Comment (kayoumido) [inline src/resources/js/components/Presentation.vue:83]: didn't the linter have a hissy fit?!

https://github.com/airbnb/javascript#strings

File: src/resources/js/components/Presentation.vue [hunk 3]
```diff
@@ -69,6 +97,28 @@ export default {
       this.message.content = content;
       this.message.type = type;
     },
+    subscribe() {
+      window.axios
+        .post(`/presentations/${this.presentation.id}/users/${this.user.id}`)
```
Comment (kayoumido) [inline src/resources/js/components/Presentation.vue:102]: https://github.com/airbnb/javascript#strings

File: src/resources/js/components/Presentation.vue [hunk 4]
```diff
@@ -69,6 +97,28 @@ export default {
...
+          this.showMessage('success', 'Félicitations vous êtes maintenant inscrit à la présentation');
+          this.user.isSubscribed = true;
+        })
+        .catch(() => {
+          this.showMessage('error', 'Oops une erreur c\'est produite lors de l\'inscription');
+        });
+    },
+    unsubscribe() {
+      window.axios
+        .delete(`/presentations/${this.presentation.id}/users/${this.user.id}`)
```
Comment (kayoumido) [inline src/resources/js/components/Presentation.vue:113]: https://github.com/airbnb/javascript#strings

**Review by kayoumido — DISMISSED** [review]
You wrote code where the linter should have done a hissy fit

**Review by kayoumido — APPROVED** [review]

**Pull request merged**